airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -155,7 +158,7 @@ private Map<UUID, DestinationRead> getDestinationReadById(final List<UUID> desti
     return destinationReads.stream().collect(Collectors.toMap(DestinationRead::getDestinationId, Function.identity()));
   }
 
-  private WebBackendConnectionRead buildWebBackendConnectionRead(final ConnectionRead connectionRead)
+  private WebBackendConnectionRead buildWebBackendConnectionRead(final ConnectionRead connectionRead, final Optional<UUID> currentSourceCatalogId)
```

Comment:
We don't need the second parameter here since it can be extracted from the first one.

Reply:
We do need this, since when there is a new schema detected, we add that newly detected schema to the connectionRead object. But we need the existing schema in order to determine whether there is a breaking schema change.